### PR TITLE
cmd: add /usr/local/* to PATH

### PR DESCRIPTION
**Pull request by zyga**
Surprisingly /usr/local plays a role in the core snap. It is there where
the image build system splices additional files that don't come from a
typical deb. This includes the fake 'apt' script but most notably the
special 'xdg-open' executable that is necessary for snapd-xdg-open to
work.

The patch also expands the existing test that checks environment to
check PATH (and SNAP and HOME while I was a it it). One regression test
also needed changes as it was measuring the stability of PATH across
invocations of snap-confine.

Fixes: https://bugs.launchpad.net/snapcraft/+bug/1661023
Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by ogra1** [comment]
looks good, land it !

File: cmd/snap-confine/snap-confine.c [hunk 1]
```diff
@@ -143,7 +143,9 @@ int main(int argc, char **argv)
 			debug
 			    ("resetting PATH to values in sync with core snap");
 			setenv("PATH",
-			       "/usr/sbin:/usr/bin:/sbin:/bin:/usr/games", 1);
+			       "/usr/local/sbin:" "/usr/sbin:" "/usr/local/bin:"
```
Comment (mvo5) [inline cmd/snap-confine/snap-confine.c]: Hm, not sure this is in sync with core:
```
$ cat /etc/environment 
PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
```
or am I misreading this?

Comment (zyga) [inline cmd/snap-confine/snap-confine.c]: Hmm, good idea to cross-check with core. Let's see...

Are those just in different order?

Comment (ogra1) [inline cmd/snap-confine/snap-confine.c]: does it matter ? as long as the /usr/local variant is always before the /usr one it should be fine i think

Comment (zyga) [inline cmd/snap-confine/snap-confine.c]: Thanks mvo, corrected now.

**Pull request merged**